app.py

```diff
@@ -1,36 +1,62 @@
+import re
+import unicodedata
+import urllib
 import flask
 
 app = flask.Flask(__name__)
 
-ZWS = flask.Markup('<span class="highlight zero-width-char">zw</span>')
+character_names = { "\n": "\\n", "\r": "\\r", "\t": "\\t" }
 
-def replace_zero_width(text):
-  text = text.replace(u'\u200B', ZWS).replace(u'\u200C', ZWS)
-  return text.replace(u'\u200D', ZWS).replace(u'\uFEFF', ZWS)
+# Load some Unicode data remotely.
+unicode_blocks = []
+for line in urllib.urlopen("http://www.unicode.org/Public/UNIDATA/Blocks.txt").read().split("\n"):
```

Comment:
Yikes, a web request on every request? Why can't we just store a local version of this file? Maybe it gets updated frequently? If that's the case I would prefer to cache the entire file (or even better, a pickle of this function's output) and have a script that updates it when needed.

At the minimum I would say remove the urlopen in this CL and perhaps add the updating logic in a future CL.